xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/impl/engine/TenantLepEngines.java

```diff
@@ -0,0 +1,98 @@
+package com.icthh.xm.commons.lep.impl.engine;
+
+import com.icthh.xm.commons.lep.api.LepEngine;
+import com.icthh.xm.commons.lep.api.LepExecutor;
+import com.icthh.xm.commons.lep.api.LepExecutorResolver;
+import com.icthh.xm.commons.lep.api.LepKey;
+import com.icthh.xm.commons.logging.util.MdcUtils;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.icthh.xm.commons.lep.impl.engine.TenantLepEngines.TenantLepEnginesStates.ACTIVE;
+import static com.icthh.xm.commons.lep.impl.engine.TenantLepEngines.TenantLepEnginesStates.DESTROYED;
+import static com.icthh.xm.commons.lep.impl.engine.TenantLepEngines.TenantLepEnginesStates.DESTROYING;
+
+@Slf4j
+@RequiredArgsConstructor
+class TenantLepEngines implements LepExecutorResolver {
```

Comment:
suggest to remove plural from name and make it `TenantLepEngime`
I think inner list of engines is internal implemetnation and should not be "communicated" to ext world.

if agree - then also rename `LepEnginesManager` to `LepEngineManager`